src/test/scala/org/constellation/domain/transaction/TransactionChainServiceTest.scala

```diff
@@ -37,17 +39,145 @@ class TransactionChainServiceTest
       last.unsafeRunSync shouldBe LastTransactionRef("", 0L)
     }
 
-    "should return last transaction ref if address is known" in {}
+    "should return last transaction ref if address is known" in {
+      val address = "DAG123"
+      val lastTx = LastTransactionRef("tx123", 1)
+      service.lastTransactionRef.set(Map(address -> lastTx)).unsafeRunSync
+
+      val last = service.getLastTransactionRef(address).unsafeRunSync
+      last shouldEqual lastTx
+    }
+  }
+
+  "getLastAcceptedTransactionRef" - {
+    "should return empty string and 0 ordinal if address is unknown" in {
+      val lastAccepted = service.getLastAcceptedTransactionRef("unknown").unsafeRunSync
+
+      lastAccepted shouldEqual LastTransactionRef("", 0L)
+    }
+
+    "should return last accepted transaction ref if address is known" in {
+      val address = "DAG123"
+      val lastTx = LastTransactionRef("tx123", 1)
+      service.lastAcceptedTransactionRef.set(Map(address -> lastTx)).unsafeRunSync
+      val lastAccepted = service.getLastAcceptedTransactionRef(address).unsafeRunSync
+
+      lastAccepted shouldEqual lastTx
```

Comment:
Why are you using `shouldEqual` instead of `shouldBe`?

Reply:
I'm using shouldEqual when the value needs to be the same (so equal) and shouldBe for example for boolean - like "value shouldBe true". Just sounds better for me. I can also use shouldBe everywhere - it should compile slightly faster.

Reply:
Yeah, we're using `shouldBe` everywhere as long as it's valid. It would be great to be consistent within the whole repository